catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/InputTransformerWaiter.java

```diff
@@ -0,0 +1,156 @@
+    this.transformerWaitTimeoutMillis = transformerWaitTimeoutMillis;
+    waitForInputTransformers();
+  }
+
+  /**
+   * Blocks and waits up to a configured ({@code org.codice.ddf.cdm.transformerWaitTimeoutSeconds}
+   * system property) amount of time for the configured {@link InputTransformer} services to be
+   * available.
+   */
+  @SuppressWarnings({
+    "squid:S1481" /* Sonar thinks variables are unused */,
+    "squid:S1854" /* Useless assignment for variable Sonar thinks is unused */
+  })
+  private void waitForInputTransformers() {
+    Set<String> configuredInputTransformers = inputTransformerIds.getIds();
+    if (configuredInputTransformers.isEmpty()) {
+      return;
+    }
+
+    Set<String> requiredInputTransformers = new HashSet<>(configuredInputTransformers);
+    LOGGER.trace(
+        "Beginning to wait for InputTransformers for {} seconds",
+        TimeUnit.MILLISECONDS.toSeconds(transformerWaitTimeoutMillis));
+
+    Failsafe.with(
+            new RetryPolicy()
+                .withMaxDuration(transformerWaitTimeoutMillis, TimeUnit.MILLISECONDS)
+                .retryWhen(false)
+                .withDelay(transformerCheckPeriodMillis, TimeUnit.MILLISECONDS))
+        .withFallback(
+            () -> {
+              throw new IllegalArgumentException(
+                  String.format(
+                      "Not all expected Input Transformers were found within %s seconds. Expected: %s, but didn't find: %s. "
+                          + "CDMs will not be able to start without all required InputTransformers. Either remove the un-found "
+                          + "IDs from the InputTransformer ID JSON files under %s, or ensure the bundles providing "
+                          + "the missing InputTransformers are started.",
+                      TimeUnit.MILLISECONDS.toSeconds(transformerWaitTimeoutMillis),
+                      configuredInputTransformers,
+                      requiredInputTransformers,
+                      inputTransformerIds.getTransformerPath()));
+            })
+        .get(
+            () -> {
+              for (ServiceReference serviceReference : inputTransformers) {
+                Object propertyValue = serviceReference.getProperty(TRANSFORMER_ID_PROPERTY);
+                if (propertyValue instanceof String) {
+                  requiredInputTransformers.remove(propertyValue);
+                  LOGGER.debug("Remaining property values {}", requiredInputTransformers);
+                }
+              }
+
+              if (requiredInputTransformers.isEmpty()) {
+                LOGGER.trace("Finished finding all InputTransformers");
+                return true;
+              }
+
+              return false;
+            });
+  }
+
+  private long getTransformerWaitTimeout() {
+    long timeoutSeconds;
+    try {
+      timeoutSeconds = Long.parseLong(System.getProperty(TRANSFORMER_WAIT_TIMEOUT_PROPERTY));
```

Comment:
Should we add the default to the System.getProperty call so we can avoid the exception if the property isn't set?

Reply:
I like the logging incase there are any issues it is easier to debug.

Reply:
I would prefer logging over adding the property systematically. I find it frustrating when I can see something is not configured, but the system acts like it is configured. It's confusing.